runtime/compiler/z/codegen/S390PrivateLinkage.cpp

```diff
@@ -2562,6 +2564,111 @@ J9::Z::PrivateLinkage::buildVirtualDispatch(TR::Node * callNode, TR::RegisterDep
+      }
+
+   TR::LabelSymbol *snippetLabel = generateLabelSymbol(cg());
+   TR::LabelSymbol *doneLabel = generateLabelSymbol(cg());
+   TR::Snippet *callSnippet = new (trHeapMemory()) TR::S390J9CallSnippetRX(cg(),
+                                                                           callNode,
+                                                                           snippetLabel,
+                                                                           doneLabel,
+                                                                           callSymRef,
+                                                                           argSize,
+                                                                           callSnippetCCDataAddress);
+   void *targetAddress = reinterpret_cast<void *>(callSnippetCCDataAddress + static_cast<intptr_t>(offsetOfSnippetOrCompiledMethod));
+   TR::StaticSymbol *snippetOrCompiledSlotSymbol = TR::StaticSymbol::createWithAddress(comp()->trHeapMemory(), TR::Address, targetAddress);
+   TR::SymbolReference *snippetOrCompiledSlotSymRef = new (comp()->trHeapMemory()) TR::SymbolReference(comp()->getSymRefTab(), snippetOrCompiledSlotSymbol, 0);
+   snippetOrCompiledSlotSymbol->setNotDataAddress();
+
+   TR::RegisterDependencyConditions *preDeps = new (trHeapMemory()) TR::RegisterDependencyConditions(dependencies->getPreConditions(), NULL,
+                                                                                                      dependencies->getAddCursorForPre(), 0, cg());
+
+   int32_t numOfAdditionalPostDeps = 0;
+   TR::Register *regRA = dependencies->searchPostConditionRegister(getReturnAddressRegister());
+   if (regRA == NULL)
+      {
+      regRA = cg()->allocateRegister();
+      numOfAdditionalPostDeps+=1;
+      }
+
+   TR::Register *regEP = dependencies->searchPostConditionRegister(getEntryPointRegister());
+   if (regEP == NULL)
+      {
+      regEP = cg()->allocateRegister();
+      numOfAdditionalPostDeps+=1;
+      }
+
+   TR::RegisterDependencyConditions *postDeps = new (trHeapMemory()) TR::RegisterDependencyConditions(dependencies, 0, numOfAdditionalPostDeps, cg());
+   postDeps->setAddCursorForPre(0);
+   postDeps->setNumPreConditions(0, trMemory());
+   postDeps->addPostConditionIfNotAlreadyInserted(regRA, getReturnAddressRegister());
+   postDeps->addPostConditionIfNotAlreadyInserted(regEP, getEntryPointRegister());
+
+   TR::Instruction *cursor = generateS390LabelInstruction(cg(), TR::InstOpCode::LABEL, callNode, generateLabelSymbol(cg()), preDeps);
+   cursor = generateRILInstruction(cg(), TR::InstOpCode::LGRL, callNode, regRA, snippetOrCompiledSlotSymRef, targetAddress, cursor);
+   cursor = generateRRInstruction(cg(), TR::InstOpCode::BASR, callNode, regRA, regRA, postDeps, cursor);
+   ((TR::S390J9CallSnippetRX *)callSnippet)->setBranchInstruction(cursor);
```

Comment:
Why not change the type of `callSnippet` to be `TR::S390J9CallSnippetRX*` at the declaration point so we can avoid this cast here?

Reply:
set branch instruction was added later so forgot to change the type. Fixed this in https://github.com/eclipse/openj9/compare/9a9d9b37daeb983c8d4e295fcc785afeecde5d78..de725983cd06ff96468a76b30321d3f8bb48caea